src/scripts/formatters.js

```diff
@@ -10,6 +10,19 @@ exports.dateFormatter = new Intl.DateTimeFormat('en-US', {
   timeZoneName: 'short'
 });
 
+// If browser doesn't support Intl (i.e. Safari), then we manually import
+// the intl polyfill and locale data.
+if (!global.Intl) {
+  require.ensure([
+    'intl',
+    'intl/locale-data/jsonp/en.js'
+  ], function (require) {
+    require('intl');
+    require('intl/locale-data/jsonp/en.js');
+  });
+}
```

Comment:
This isn’t actually working for me. I’m pretty sure there are two main problems here:

1. This polyfill import comes *after* the first use of `Intl` (line 3), which is where I get an error. (Note this issue is only in Safari 9.1 and earlier, which is… kinda old.)
2. The import would be asynchronous anyway, so wouldn’t be available until after the `exports.dateFormatter = new Intl...` line ran even if it was written before it.

I think the best way to resolve this would be to move the polyfill loading up to [`src/scripts/main.jsx`](https://github.com/edgi-govdata-archiving/web-monitoring-ui/blob/master/src/scripts/main.jsx) so that it delays the entire application from loading if it’s needed like so:

```js
import '../css/styles.css';
import '../css/diff.css';

function startApp () {
  // These dependencies assume Intl will be present, so they can't be imported until
  // we get here, where we know the runtime environment is polyfilled if necessary.
  const React = require('react');
  const ReactDOM = require('react-dom');
  const WebMonitoringUi = require('../components/web-monitoring-ui').default;

  ReactDOM.render(
    <WebMonitoringUi />,
    document.getElementById('web-monitoring-ui-root')
  );
}

// If browser doesn't support Intl (i.e. Safari 9.1 and earlier), then we
// manually import the intl polyfill and locale data.
if (window.Intl) {
  startApp();
}
else {
  require.ensure(['intl', 'intl/locale-data/jsonp/en.js'], require => {
    require('intl');
    require('intl/locale-data/jsonp/en.js');

    startApp();
  })
}
```

You’ll also need to update [the deploy script at `scripts/heroku-deploy.sh`](https://github.com/edgi-govdata-archiving/web-monitoring-ui/blob/master/scripts/heroku-deploy.sh#L27) to include the new `.js` file this will output with the `Intl` polyfill. (That probably just means changing `dist/bundle.*` there to `dist/*.js*`).

---

**Side note:** now that you’ve dug into this, I’m realizing all the above could be a little over-complicated. 😕

Back when we wrote #155, Safari 9 (the latest version without `Intl` support) wasn’t totally uncommon. That’s less true now. A simpler solution might be to drop the polyfill and just replace line 3 of this file with something like (warning: totally untested code!):

```js
if (window.Intl) {
  exports.dateFormatter = new Intl.DateTimeFormat(/* same args as we are already using */);
}
else {
  exports.dateFormatter = {
    format (date) {
      return date.toLocaleString();
    }
  }
}
```

You won’t necessarily quite get the format you want all the time (when it winds up falling back to `toLocaleString()`), but it will happen rarely, you won’t have to load extra code, and it’ll still get you a readable date string.

Reply:
I went ahead and did this.